From be03ec66f116cf8828ff960abbe9cedc0c7f1373 Mon Sep 17 00:00:00 2001
From: Nava kishore Manne <nava.manne@xilinx.com>
Date: Mon, 6 Apr 2020 18:52:49 +0530
Subject: [PATCH 1319/1852] fpga: versal: Use 32-bit DMA addressing

commit c0891b693eef77bf57400f9c23c1143a94224653 from
https://github.com/Xilinx/linux-xlnx.git

PLM firmware uses PMC DMA to load/process the PDI image.
The PMC DMA currently supporting only 32-bit operation.So
This patch changes the 44 bit DMA mask to 32 bit.

Signed-off-by: Nava kishore Manne <nava.manne@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/fpga/versal-fpga.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/fpga/versal-fpga.c b/drivers/fpga/versal-fpga.c
index 791d91eb0bb8..a6694d7fd27a 100644
--- a/drivers/fpga/versal-fpga.c
+++ b/drivers/fpga/versal-fpga.c
@@ -96,7 +96,7 @@ static int versal_fpga_probe(struct platform_device *pdev)
 		return -ENOMEM;
 
 	priv->dev = dev;
-	ret = dma_set_mask_and_coherent(&pdev->dev, DMA_BIT_MASK(44));
+	ret = dma_set_mask_and_coherent(&pdev->dev, DMA_BIT_MASK(32));
 	if (ret < 0) {
 		dev_err(dev, "no usable DMA configuration");
 		return ret;
-- 
2.31.1

